[doc] [client] [go] Add chunking to go-client doc (#17789)
apache/pulsar: site2/docs/client-libraries-go.md

}
```

#### How to use chunking in producer

```go

> client, err := NewClient(pulsar.ClientOptions{

Reviewer: ```suggestion
client, err := pulsar.NewClient(pulsar.ClientOptions{
```

	URL: serviceURL,
})

if err != nil {
	log.Fatal(err)
}
defer client.Close()

// The message chunking feature is OFF by default.

> // By default, producer chunks the large message based on max message size (maxMessageSize) configured at broker (eg: 5MB).

Reviewer 2: ```suggestion
// By default, a producer chunks the large message based on the max message size (`maxMessageSize`) configured at the broker side (for example, 5MB).
```

Author: Good suggestion, thx!

> // Client can also configure max chunked size using producer configuration ChunkMaxMessageSize.

Reviewer 2: ```suggestion
// Client can also configure the max chunked size using the producer configuration `ChunkMaxMessageSize`.
```

Author: Good suggestion, thx!

> // Attention, to enable chunking, you need to disable batching (DisableBatching=true) concurrently.

Reviewer 2: ```suggestion
// Note: to enable chunking, you need to disable batching (`DisableBatching=true`) concurrently.
```

Author: Good suggestion, thx!
